packages/router/useRouteParams/index.ts

```diff
@@ -57,9 +57,10 @@ export function useRouteParams<
       _params.set(name, (v === defaultValue || v === null) ? undefined : v)
 
       trigger()
-
       nextTick(() => {
-        router[toValue(mode)]({ ...route, params: { ...route.params, ...Object.fromEntries(_params.entries()) } })
+        const { name: pathName, path, ...otherRouteParams } = route
+        const routeParams = { name: pathName, ...otherRouteParams }
+        router[toValue(mode)]({ ...routeParams, params: { ...route.params, ...Object.fromEntries(_params.entries()) } } as RouteLocationRaw)
       })
```

Comment:
I think we can just remove the `...route` and it will work well. What do you think?

```ts
nextTick(() => {
  router[toValue(mode)]({
    params: {
      ...route.params,
      ...Object.fromEntries(_params.entries()),
    },
  });
});
```

Reply:
@antfu any thoughts?

Reply:
I think it is necessary to keep the parameters other than the path in the route, otherwise we can't get some parameters in the next level of routing, such as meta

> I think we can just remove the `...route` and it will work well. What do you think?
> 
> ```ts
> nextTick(() => {
>   router[toValue(mode)]({
>     params: {
>       ...route.params,
>       ...Object.fromEntries(_params.entries()),
>     },
>   });
> });
> ```

I think the route parameter should be needed so that the attributes in the current route such as meta can be obtained after the jump

Reply:
We only need to remove the path parameter, because vue router does not support params and path together